modules/Connection.py

```diff
@@ -1,74 +1,77 @@
+import logging
+
+LOG = logging.getLogger('vcsa-exporter')
 
 
 class Connection:
-    def login(target, user, password):
+    def __init__(self, vcenter):
+        self.target = vcenter.name
+        self.user = vcenter.user
+        self.pw = vcenter.pw
+        self.session_id = None
+
+    def login(self):
         disable_warnings(exceptions.InsecureRequestWarning)
-        if os.environ['DEBUG'] == "1":
-            print("login", target)
-        url = "https://" + target + "/rest/com/vmware/cis/session" 
+        LOG.debug(f"login {self.target}")
+        url = "https://" + self.target + "/rest/com/vmware/cis/session"
         try:
-            response = requests.post(url, auth=(user, password), verify=False)
+            response = requests.post(url, auth=(self.user, self.pw), verify=False)
         except Exception as e:
-            print("Problem connecting to", target, "Error:", str(e))
+            LOG.error(f"Problem connecting to {self.target} Error: {str(e)}")
             return False
-        
         if response.status_code == 200:
-            return response.json()['value']
+            self.session_id = response.json()['value']
+            LOG.debug(f"login session id {self.target} {self.session_id}")
         else:
-            print("Problem logging into", target, ":", response.text)
-    
-    def get_request(target, key, session_id):
+            LOG.warning(f"Problem logging into {self.target}: {response.text}")
+            return False
+
+    def get_request(self, key):
         disable_warnings(exceptions.InsecureRequestWarning)
-        if os.environ['DEBUG'] == "1":
-            print("request",target, key)
-        url = "https://" + target + "/rest/" + key 
+        LOG.debug(f"request {self.target}, {key}")
+        url = "https://" + self.target + "/rest/" + key
         try:
             response = requests.get(url, verify=False,
-                                    headers={"vmware-api-session-id": session_id})
+                                    headers={"vmware-api-session-id": self.session_id})
         except Exception as e:
-            print("Problem handling", key, "for", target, ":", str(e))
+            LOG.error(f"Problem handling {key} for {self.target}: {str(e)}")
             return False
         if response.status_code == 200:
             return response.json()
         else:
-            print("Problem with get return of", target, ":", response.text)
+            LOG.warning(f"Problem with get return of {self.target}: {response.text}")
             return False
 
-    def post_request(target, key, data, session_id):
+    def post_request(self, key, data):
         disable_warnings(exceptions.InsecureRequestWarning)
-        if os.environ['DEBUG'] == "1":
-            print("request", target, key)
-        url = "https://" + target + "/rest/" + key + '?' + data
+        LOG.debug(f"request {self.target} {key}")
+        url = "https://" + self.target + "/rest/" + key + '?' + data
```

Comment:
Maybe better: 
```suggestion
        url = "https://" + self.target + "/rest/" + key
        try:
           response = requests.post(url, verify=False, data=data, headers={"vmware-api-session-id": session_id})	            
```

Reply:
moved it back, won't work with data in there straight.